workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfigurationEditor.cs

```diff
@@ -73,81 +92,408 @@ public override void OnInspectorGUI()
         private bool DrawWorkerConfiguration(WorkerBuildConfiguration configurationForWorker)
         {
             var workerType = configurationForWorker.WorkerType;
+            var expanded = ExpandedFoldouts.Contains(configurationForWorker.WorkerType);
+            bool currentExpanded;
 
-            EditorGUILayout.BeginHorizontal();
-            configurationForWorker.ShowFoldout =
-                EditorGUILayout.Foldout(configurationForWorker.ShowFoldout, workerType);
-
-            var controlRect = EditorGUILayout.GetControlRect(false);
-            var buttonRect = new Rect(controlRect);
-            buttonRect.width = buttonRect.height + 5;
-            if (GUI.Button(buttonRect, RemoveWorkerTypeButtonContents))
+            using (new EditorGUILayout.HorizontalScope())
             {
-                return false;
+                currentExpanded =
+                    EditorGUILayout.Foldout(expanded, workerType);
+
+                GUILayout.FlexibleSpace();
+                if (GUILayout.Button(RemoveWorkerTypeButtonContents, EditorStyles.miniButton))
+                {
+                    ExpandedFoldouts.Remove(configurationForWorker.WorkerType);
+                    return false;
+                }
             }
 
-            EditorGUILayout.EndHorizontal();
+            if (expanded && !currentExpanded)
+            {
+                ExpandedFoldouts.Remove(configurationForWorker.WorkerType);
+            }
+            else if (!expanded && currentExpanded)
+            {
+                ExpandedFoldouts.Add(configurationForWorker.WorkerType);
+            }
 
-            if (configurationForWorker.ShowFoldout)
+            using (IndentLevelScope(1))
             {
-                using (IndentLevelScope(1))
+                if (currentExpanded)
                 {
                     DrawScenesInspectorForWorker(configurationForWorker);
                     DrawEnvironmentInspectorForWorker(configurationForWorker);
                 }
+                else
+                {
+                    // Draw an overview of issues.
+                    if (GetConfigurationProblems(BuildEnvironment.Local, configurationForWorker).Count > 0 ||
+                        GetConfigurationProblems(BuildEnvironment.Cloud, configurationForWorker).Count > 0)
+                    {
+                        EditorGUILayout.HelpBox("Problems found", MessageType.Warning);
```

Comment:
Does it print out the problems elsewhere?

Reply:
The individual problems are drawn in-line once the workers are expanded.
I've also refactored the WorkerBuilder to use the same code path to validate and print to the console during a build.